src/plugins/usage_collection/server/routes/report_metrics.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { METRIC_TYPE, Report } from '@kbn/analytics';
+import { IRouter } from '../../../../../src/core/server';
+import { storeReport } from '../store_report';
+
+export function registerUiMetricRoute(router: IRouter, getLegacySavedObjects: () => any) {
+  router.post(
+    {
+      path: '/api/ui_metric/report',
+      validate: {
```

Comment:
The schema validation's rather long, can we please move it into a `schema` folder, perhaps in a `post_report` file?